internal/pkg/api/handleCheckin.go

```diff
@@ -454,22 +455,21 @@ func (ct *CheckinT) processUpgradeDetails(ctx context.Context, agent *model.Agen
 		return nil
 	}
 	// update docs with in progress details
-
 	vSpan, vCtx := apm.StartSpan(ctx, "Check update action", "validate")
-	action, err := ct.verifyActionExists(ctx, vSpan, agent, details)
-	if err != nil {
+
+	var actionTraceparent string
+	if action, err := ct.verifyActionExists(ctx, vSpan, agent, details); err == nil && action != nil {
```

Comment:
Does Kibana tolerate having an action ID it doesn't recognize as well?

I think passing through the upgrade details unmodified is correct, but we need to avoid Kibana now or in the future trying to do this same validation or making assumption about this field.

Reply:
> I think passing through the upgrade details unmodified is correct

Well a possibility here would be removing the action_id if it doesn't exist so kibana can't depend on it, or putting some marker in the details about whether the action exists or not (though kibana can also just check this).

Reply:
@juliaElastic any opinions on if sending upgrade_details to Kibana for action_ids that don't exist or are from another cluster is a problem?